### DISCUSSION TITLE
Native passkey experience on Apple platforms

### DISCUSSION THREAD

**Discussion by groue** [ory/kratos#3038] [Q&A] [answered]
Hello,

Apple platforms offer support for passkeys in browsers, but also in native applications. This question is about Ory Kratos and native macOS/iOS apps.

I can read in [Ory documentation](https://www.ory.sh/docs/kratos/mfa/webauthn-fido-yubikey#limitations):

> WebAuthn is a browser-only standard. It doesn't work with native mobile apps.

Is this limitation still accurate?

If so, is there any plan to lift it?

Any advice from other experienced user?

Thank you very much in advance!

<details><summary>A short description of the mismatch between Kratos apis and Apple SDK</summary>

The [native Apple SDK](https://developer.apple.com/documentation/authenticationservices/public-private_key_authentication) supports various scenarios involving passkeys: registration, authentication, etc.

There is a [sample application](https://developer.apple.com/documentation/authenticationservices/connecting_to_a_service_with_passkeys), as well as a [Supporting passkeys](https://developer.apple.com/documentation/authenticationservices/public-private_key_authentication/supporting_passkeys) guide.

Most native apis require a _server challenge_. And I don't find in the [Ory Identities API](https://www.ory.sh/docs/kratos/reference/api) any endpoint that provides such a challenge:

- `/.well-known/ory/webauthn.js` is for Javascript applications.
- `/self-service/login/api` does not return any WebAuthn challenge.

So it indeed looks like it is impossible to leverage the Ory Kratos support for passkeys from a native iOS or macOS application.

I wish I'm wrong and I was just unable to navigate the documentation correctly!

</details>

**Comment by thomastthai** [comment]
I was looking for similar answers and implementations in a [previous discussion](https://github.com/ory/kratos/discussions/2967). So far, nothing concrete yet.

**Comment by kmherrmann** [comment]
Thanks for the question. Yes, the limitation is still current, and yes, we have plans to lift it and will ship improvements to our API-based flows in the coming months - including full Webauthn support.

**Reply by groue** [reply]
Thank you very much @kmherrmann for lifting the doubt 👍

**Comment by kmherrmann** [comment]
This is now in development and should become generally available in the
next few weeks.

**Comment by vinckr** [comment, marked as answer]
It is released now @groue !
https://changelog.ory.sh/announcements/native-social-sign-in-with-apple

**Reply by groue** [reply]
Thank you, and congratulations 🎉🙂

**Comment by ajcampbell** [comment]
Thank you very much. Sent from my iPhoneOn 29 Sep 2023, at 08:49, Vincent ***@***.***> wrote:﻿
It is released now @groue !
https://changelog.ory.sh/announcements/native-social-sign-in-with-apple

**Comment by slashmo** [comment]
Hey @vinckr, thanks for working on better native integrations 👋
I'm a bit confused about the status of native passkey support. As far as I understand the comment above (https://github.com/ory/kratos/discussions/3038#discussioncomment-7141853) and the original question it should be available now. The linked changelog entry is about Sign in with Apple though, not passkeys. For native passkeys we'll still need the challenge parameter on iOS to pass to Apple's SDK: https://developer.apple.com/documentation/authenticationservices/public-private_key_authentication/supporting_passkeys
Is this available in Ory Cloud already, still a work in progress, or not yet started?
